### ISSUE TITLE
Consider a11y enhancements for date components

### ISSUE THREAD

**Issue by bkjohnson** [department-of-veterans-affairs/vets-design-system-documentation#1071]
## Description
For `<va-date monthYearOnly />` and `<va-memorable-date>` we had some related items under the "Consider" portion of the a11y feedback.

### Tickets

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/39727
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/844

### Items to consider

- Consider adding context to the error message to let users know which exact fields are producing an error
- (`<va-memorable-date>`) The attribute aria-describedby="dateHint" is included on all elements, but is not read by the screen reader for each group that includes the label + input field. I would recommend either of the following:
    - Removing the aria-describedby attributes, and instead, including text for screen readers within the , specific to each input field;
    - Including the aria-describedby attributes on each input element with helpful text specific to each input field
- Think about the UX when there are multiple error messages appearing on the page
- Consider adding responsive styles

@davidakennedy @TiffanyPender please let me know if I've missed anything.

**Comment by TiffanyPender** [comment]
@bkjohnson This looks like it captures all of the items to consider for these components. 👍

**Comment by davidakennedy** [comment]
Following up on this to tie some issues together. @jaredcunha filed a very similar issue in #1337 that calls out many of the same things.

From Jared's issue:

> Hint text does not seem to have a programatic link to the hint text

We reference this above, but it looks like it's not consistent on the [date component](https://design.va.gov/components/form/date-input). The attribute aria-describedby="dateHint" is on [memorable date](https://design.va.gov/components/form/memorable-date) but not on [Month Year](https://design.va.gov/components/form/date-input#monthyear). We should sync up approaches there.

> The select boxes have an aria-label of "Please enter two digits for the month/day". With the current select box, this has a couple related issues for screen reader useres:
> 
> 1. Month uses month name, not numbers, so typing numbers does not work here.
> 2. The text instructs user to enter two digits for day, but some days are a single digit.
> 3. If the user does not select a month first, the listbox is empty. The user might benefit from some instructions to select a month first before selecting a day if they've skipped the month field.
> 4. This seems to have mixed results depending on how you navigate the page. Tab navigation, VoiceOver announces "Month, menu pop up collapsed, Please enter two digits for the month". In keyboard navigation controls, reads "Group, please enter two digits for the month" before digging further into the component to get to the dropdown.

I think if we approach removing the aria-label attributes for the fields that most of the issues go away. Numbers 3 and 4 should be tested as we determine an approach to make sure those are improved. It's worth tacking them on here since they're all related.

**Comment by jaredcunha** [comment]
@davidakennedy I think you've summarized my issue well here. 

Perhaps, while at it, whatever any kind of hint text becomes, can we also address the use of "Please" by not using it?

**Comment by humancompanion-usds** [comment]
@jeana-adhoc - Can you have a look at this one and determine if there is any work left to be done here? Thanks!

**Comment by jeana-adhoc** [comment]
**NOTES - WIP**

Custom error message props are available in the Memorable date component
- `customDayErrorMessage` "Please select a month"
- `customMonthErrorMessage` "Please enter a day between 1 and 30" if a month is selected it provides the right number of days
- `customYearErrorMessage` "Please enter a year between 1900 and 2125"

Custom error message props are available in the date component
- `invalidMonth`"Please select a month" 
- `invalidDay` "Please enter a day between 1 and 28" - if a month is selected it provides the right number of days
- `invalidYear` "Please enter a year between 1900 and 2125"


🛑  Remaining Issues & Recommendations
- In `va-date`, the hint text is not associated with the label. It is currently in a span outside of the legend, and not associated with any field. 
  -  @davidakennedy 's suggestion above should fix this "We reference this above, but it looks like it's not consistent on the [date component](https://design.va.gov/components/form/date-input). The attribute aria-describedby="dateHint" is on [memorable date](https://design.va.gov/components/form/memorable-date) but not on [Month Year](https://design.va.gov/components/form/date-input#monthyear). We should sync up approaches there."
- Baked in error messages should be updated to follow the [VA Content Style guide](https://design.va.gov/patterns/ask-users-for/dates#content-considerations) - Updates should be handled in the component, and then in the forms library if appropriate.
- In the default of Memorable date, the hint text still says "Enter 2 digits for the month and day and 4 digits for the year". The version with the select drop downs, the hint text says "For example: January 19, 2000".  I think the select version is ok, but we may want to improve the default version

**Comment by jeana-adhoc** [comment]
Next steps:
1. Review content suggestions above with content folks
2. Work with Amy to see if we can build better hint text for screen readers.
